### PR TITLE
[qob] Use one batch for the lifetime of a hl.ServiceBackend

### PR DESCRIPTION
**Pull request by daniel-goldstein**
CHANGELOG: When using Query-on-Batch, all jobs for a single hail session are inserted into the same batch instead of one batch per action.

### CODE REVIEW

**Comment by daniel-goldstein** [comment]
This is pseudo-stacked on #12490 in that I made a change here to serialize `_async_execute_many` but in #12490 I remove it altogether which takes care of the complications of multiple concurrent Query Drivers in the same batch.

File: hail/python/hail/backend/service_backend.py [hunk 1]
```diff
@@ -325,43 +325,50 @@ async def _rpc(self,
...
                         deploy_config = get_deploy_config()
-                        url = deploy_config.external_url('batch', f'/batches/{b.id}/jobs/1')
-                        print(f'Submitted batch {b.id}, see {url}')
-
-                    await b.wait(description=name,
-                                 disable_progress_bar=self.disable_progress_bar,
-                                 progress=progress)
+                        # FIXME Should we not `print`?
+                        url = deploy_config.external_url('batch', f'/batches/{self._batch.id}/jobs/{j.job_id}')
+                        print(f'Submitted to batch {self._batch.id}, see {url}')
```
Comment (danking) [inline hail/python/hail/backend/service_backend.py:360]: I'm somewhat inclined to not print. The progress bar should provide all necessary information

**Review by danking — COMMENTED** [review]
LGTM

File: hail/python/hail/backend/service_backend.py (hunk 1 above)
Comment (daniel-goldstein) [reply hail/python/hail/backend/service_backend.py:360]: Happy to axe that.

**Pull request merged**